plugins/modules/selinux.py

```diff
@@ -150,6 +174,17 @@ def set_state(module, state):
         module.fail_json(msg=msg)
 
 
+def set_kernel_enabled(module, value):
+    rc, stdout, stderr = module.run_command(['grubby', '--update-kernel=ALL',
+                                             '--args' if value else '--remove-args',
+                                             'selinux=0'])
```

Comment:
maybe I'm misunderstand, but this looks like it's disabling selinux instead of enabling

Reply:
Yes, you're right. I got it backwards both here and in the function call, so it just happened to work correctly. I'll update the code to make more sense...

Reply:
Sorry if I missed something but `grubby` makes change the behavior (for example, modifying /etc/default/grub file on RHEL8) of the selinux module beyond what the user was aware of before. So I think that we needed to mention these changes in the docs.

* https://github.com/ansible-collections/ansible.posix/blob/33e33c0af0738718170a8d85eed00d632f6490ff/plugins/modules/selinux.py#L15-L18
* https://github.com/ansible-collections/ansible.posix/blob/33e33c0af0738718170a8d85eed00d632f6490ff/plugins/modules/selinux.py#L25-L30

Also, if `selinux=0` is set by grubby, I think it is better to include this change in the result:
* https://github.com/ansible-collections/ansible.posix/blob/33e33c0af0738718170a8d85eed00d632f6490ff/plugins/modules/selinux.py#L58-L84

Reply:
Indeed, I'm now adding a new module param for changing the kernel config, which will definitely need a documentation update.

As for including the change in the result, what exactly should be returned there? The final state after the change (disabled vs. enabled)? When no change is made to kernel config, should it not return that field or instead return some "undefined" value? The kernel config state may be inconsistent or failed to be detected, in which case it's unclear what should be returned as the final state.

Reply:
Hi, thank you for your clarification.
Yeah, if you add an option like `update_bootloader` and document its behavior, users understand the behavior of this option that the change will be made when the task is successful.
Also, if grubby update fails, the task will be failed with the message. So in this case I don't think it's necessary to include the changes in the return value.